shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm

```diff
@@ -113,7 +113,7 @@ - (void)testInfoPlist {
 
 - (void)testDeallocated {
   __weak FlutterEngine* weakEngine = nil;
-  {
+  @autoreleasepool {
```

Comment:
Were you able to track down how the engine is getting onto the autorelease pool?

Reply:
No. I'm not sure how to do that. 

> I have some concerns that the issue is another thread that has a retain on the engine
In what situation could this happen?

Reply:
This is only happening when running the test locally tho, ci never failed. So Im not sure what the difference is that caused the FlutterEngine getting onto the pool.